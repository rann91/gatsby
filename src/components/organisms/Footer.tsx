import { graphql, useStaticQuery } from 'gatsby'
import React from 'react'
import {
  Strapi_Footer,
  Strapi_FooterContentDynamicZone
} from '../../typings/graphql'
import Bottom from '../molecules/Bottom'
import Content from '../templates/Content'

interface Data {
  strapi: {
    footer: Strapi_Footer
  }
}

const Footer = () => {
  const data = useStaticQuery<Data>(graphql`
    query {
      strapi {
        footer {
          id
          content {
            ...Cta
          }
        }
      }
    }
  `)

  return (
    <footer>
      <Content
        components={
          (data.strapi.footer.content ||
            []) as Strapi_FooterContentDynamicZone[]
        }
      />
      <Bottom />
    </footer>
  )
}

export default Footer
