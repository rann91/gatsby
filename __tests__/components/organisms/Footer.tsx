import { render } from '@testing-library/react'
import { useStaticQuery } from 'gatsby'
import React from 'react'
import Footer from '../../../src/components/organisms/Footer'
import data from '../../fixtures/footer.json'

beforeEach(() => {
  ;(useStaticQuery as jest.Mock).mockImplementationOnce(() => data)
})

describe('Footer', () => {
  it('renders correctly', () => {
    const { container } = render(<Footer />)

    expect(container).toMatchSnapshot()
  })
})
