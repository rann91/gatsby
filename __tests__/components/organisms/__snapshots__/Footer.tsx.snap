// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders correctly 1`] = `
<div>
  <footer>
    <section
      class="css-fvc22l-Cta"
      data-testid="section"
    >
      <div
        class="css-1yrmf69"
      >
        <div
          class="css-i1z8pl"
        >
          <div
            class="css-ycxng9-TextContainer"
            data-testid="text-container"
          >
            <h2
              class="css-1wf7gym"
            >
              Grow your business
            </h2>
            <p
              class="css-w9nyfe-Cta"
            >
              Today is the day to build the business of your dreams. Share your mission with the world â€” and blow your customers away.
            </p>
            <a
              class="css-1s79t5o-Cta"
              data-testid="gatsby-link"
              href="/about"
              target="_self"
            >
              Learn more
            </a>
          </div>
        </div>
      </div>
    </section>
    <div
      class="css-1ca5803"
      data-testid="bottom"
    >
      <div
        class="css-1yrmf69"
      >
        <p
          class="css-10im984-Bottom"
        >
          Made with
           
          <a
            class="css-vurnku"
            href="https://www.gatsbyjs.org"
            rel="noopener noreferrer"
            target="_blank"
          >
            Gatsby
          </a>
           
          and
           
          <a
            class="css-vurnku"
            href="https://strapi.io"
            rel="noopener noreferrer"
            target="_blank"
          >
            Strapi
          </a>
          . Design by
           
          <a
            class="css-vurnku"
            href="https://webflow.com"
            rel="noopener noreferrer"
            target="_blank"
          >
            Webflow
          </a>
          .
        </p>
      </div>
    </div>
  </footer>
</div>
`;
